.github/workflows/build-plugin-archive.yml

```diff
@@ -0,0 +1,219 @@
+name: Create plugin archive
+
+on:
+  workflow_call:
+    inputs:
+      PLUGIN_MAIN_FILE:
+        description: The path/name of the plugin main file.
+        required: false
+        default: './index.php'
+        type: string
+      PLUGIN_VERSION:
+        description: The plugin version.
+        required: true
+        type: string
+      COMPOSER_ARGS_DEV:
+        description: Set of arguments passed to Composer for build setup.
+        default: '--prefer-dist'
+        required: false
+        type: string
+      COMPOSER_ARGS:
+        description: Set of arguments passed to Composer for production setup.
+        default: '--no-dev --no-scripts --prefer-dist --optimize-autoloader'
+        required: false
+        type: string
+      NPM_REGISTRY_DOMAIN:
+        description: Domain of the private npm registry.
+        default: https://npm.pkg.github.com/
+        required: false
+        type: string
+      NODE_VERSION:
+        description: Node version with which the unit tests are to be executed.
+        default: 16
+        required: false
+        type: string
+      EXTRA_PHP_FILE:
+        description: Path to a custom php script to run before creating the release archive.
+        default: ''
+        required: false
+        type: string
+      EXTRA_PHP:
+        description: Run custom php code before creating the release archive.
+        default: ''
+        required: false
+        type: string
+    secrets:
+      COMPOSER_AUTH_JSON:
+        description: Authentication for privately hosted packages and repositories as a JSON formatted object.
+        required: false
+      NPM_REGISTRY_TOKEN:
+        description: Authentication for the private npm registry.
+        required: false
+      ENV_VARS:
+        description: Additional environment variables as a JSON formatted object.
+        required: false
+
+jobs:
+  create-plugin-archive:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    env:
+      COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+      NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+    outputs:
+      artifact: ${{ steps.set-artifact-name.outputs.artifact }}
+    steps:
+
+      #
+      # Project setup
+      #
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Create plugin basename from specified main file
+        id: basename
+        run: echo "basename=$(basename ${{ inputs.PLUGIN_MAIN_FILE }} .php)" >> $GITHUB_OUTPUT
```

Comment:
Would be great to be able to add a prefix. Like `JIRA-123`.

https://github.com/woocommerce/woocommerce-paypal-payments/blob/664a53dd36ed70b68745bd21552b0160d7b58462/.github/workflows/package.yml#L46-L50

Reply:
Instead of relying on the name of the plugin main file, there is now a new input called `PLUGIN_BASENAME`.
If this is not passed, the workflow will fall back to the name of the repository.